runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -172,20 +179,41 @@ private void restoreTable(Path filePath, UUID streamId) throws IOException {
      */
     private boolean isTableTagged(String tableBackup) {
         // tableBackup name format: uuid.namespace$tableName
+        UUID uuid = UUID.fromString(tableBackup.substring(0, tableBackup.indexOf(".")));
+        if (!tableTagged.isEmpty()) {
+            return tableTagged.get(uuid);
+        }
+
         String[] strings = tableBackup.substring(tableBackup.indexOf(".")+1).split("\\$");
         if (Objects.equals(strings[0], TableRegistry.CORFU_SYSTEM_NAMESPACE) &&
                 Objects.equals(strings[1], TableRegistry.REGISTRY_TABLE_NAME)) {
             return true;
         }
 
-        CorfuStoreMetadata.TableName tableName = CorfuStoreMetadata.TableName.newBuilder()
-                .setNamespace(strings[0]).setTableName(strings[1]).build();
-        return rt.getTableRegistry()
-                .getRegistryTable()
-                .get(tableName)
-                .getMetadata()
-                .getTableOptions()
-                .getRequiresBackupSupport();
+        // Populate tableTagged map
+        // The reason we don't read the tag by getRegistryTable().get(tableName)
+        // is that, the tableName extracted from tableBackup is not guaranteed
+        // to be a full name. 'TarArchiveOutputStream.LONGFILE_TRUNCATE' is used
+        // to generate the backup files which truncates the table names which are
+        // too long. The uuid as the first part of the file name is always complete
+        // since it has fixed length and is smaller than the truncate limit (100).
+        try {
+            rt.getTableRegistry()
+                    .getRegistryTable()
+                    .forEach((key, value) -> {
```

Comment:
While it is fine for this use - should we consider entryStream() instead of directly iterating over any CorfuTable - the reason is some tables maybe memory backed in which case I am not sure if the Map abstraction will hold the same way?

Reply:
I see. Plus that the new CorfuTable in MVCC no longer provides forEach. I changed to `entryStream().forEach()`